src/main/scala/loamstream/loam/LoamGraphValidation.scala

```diff
@@ -140,7 +141,7 @@ object LoamGraphValidation {
     }
   }
 
-  val eachToolHasEitherInputsOrOutputs: LoamToolRule[Unit] = new LoamToolRule[Unit] {
+  object eachToolHasEitherInputsOrOutputs extends LoamToolRule[Unit] {
```

Comment:
Other than "flavor of the month" coding style (which I totally do and have nothing against), is there a reason to change all these variables to objects? Was there an advantage elsewhere in the PR that I missed?

Note: I'm not against it, I prefer "globals" as singleton objects over `val`, but want to make sure I'm not missing something.

Reply:
Yes, I made these objects so their classes would get meaningful names, instead of `$$anon$15` or whatever.  All those anonymous class names made it so I couldn't find out what part of `LoamGraphValidation` (a big file) was slow when looking at profiler data.  

Once I finished profiling, I just left this alone.  There are downsides/tradeoffs to objects over anonymous classes, namely laziness and more bytecode for lazy initialization, but once this worked, I didn't mess with it further.